ui/src/shared/utils/vis.ts

```diff
@@ -282,6 +297,34 @@ export const defaultYColumn = (
   return null
 }
 
+export const mosaicYcolumn = (
+  table: Table,
+  preferredColumnKey?: string
+): string | null => {
+  const validColumnKeys = getStringColumns(table)
+  if (validColumnKeys.includes(preferredColumnKey)) {
+    return preferredColumnKey
+  }
+
+  const invalidMosaicYColumns = new Set([
```

Comment:
why a `Set` and not a regular array?

Reply:
I don't think there was a particular reason for it. We just happened to use a set here.

Reply:
Is it better to use a regular array than a Set?

Reply:
Groupable columns used set, so I was basing it off of that.

Reply:
i don't think it's better or worse, i think using a [`Set`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set) implies that you care about uniqueness.

Here's the description from MDN:
>Set objects are collections of values. You can iterate through the elements of a set in insertion order. A value in the Set may only occur once; it is unique in the Set's collection.